### PR TITLE
Support limiting the content length of user profile #1036

### PR DESCRIPTION
**Pull request by DevPJ9**
Added limits to the mentioned fields. 

- name: 20
- intro: 100
- profilePicture: 100

### CODE REVIEW

File: turms-server-common/src/main/java/im/turms/server/common/infra/property/env/service/business/user/UserProperties.java [hunk 1]
```diff
@@ -58,4 +60,21 @@ public class UserProperties {
     @MutableProperty
     private boolean activateUserWhenAdded = true;
 
+    @Description("Maximum length of the name a user can enter")
```
Comment (JamesChenX) [inline turms-server-common/src/main/java/im/turms/server/common/infra/property/env/service/business/user/UserProperties.java:63]: `The maximum allowed length for a user's name` seems better.

File: turms-server-common/src/main/java/im/turms/server/common/infra/property/env/service/business/user/UserProperties.java [hunk 2]
```diff
@@ -58,4 +60,21 @@ public class UserProperties {
...
+    @GlobalProperty
+    @MutableProperty
+    @Min(0)
+    private int nameLengthLimit = 20;
+
+    @Description("Maximum length of the intro a user can enter")
```
Comment (JamesChenX) [inline turms-server-common/src/main/java/im/turms/server/common/infra/property/env/service/business/user/UserProperties.java:69]: `The maximum allowed length for a user's intro` seems better.

File: turms-server-common/src/main/java/im/turms/server/common/infra/property/env/service/business/user/UserProperties.java [hunk 3]
```diff
@@ -58,4 +60,21 @@ public class UserProperties {
...
+    @GlobalProperty
+    @MutableProperty
+    @Min(0)
+    private int introLengthLimit = 100;
+
+    @Description("Maximum length of the profile picture a user can enter")
```
Comment (JamesChenX) [inline turms-server-common/src/main/java/im/turms/server/common/infra/property/env/service/business/user/UserProperties.java:75]: `The maximum allowed length for a user's profile picture` seems better.

**Comment by JamesChenX** [comment]
Thanks for your PR!
There are still some changes that are needed:
1. You have added the properties, but you haven't applied them, so these are properties are useless currently.
So you need to 1. use these properties in code; 2. update these properties when the global properties change. And you can reference the implementation of `im.turms.server.common.infra.property.env.service.business.user.FriendRequestProperties#contentLimit` because their implementation should be very similar.
image
2. Update the description of properties as mentioned in the suggestions.

**Comment by DevPJ9** [comment]
> Thanks for your PR! There are still some changes that are needed:
> 
> 1. You have added the properties, but you haven't applied them, so these are properties are useless currently.
>    So you need to 1. use these properties in code; 2. update these properties when the global properties change. And you can reference the implementation of `im.turms.server.common.infra.property.env.service.business.user.FriendRequestProperties#contentLimit` because their implementation should be very similar.
>    image
> 2. Update the description of properties as mentioned in the suggestions.

I have done the changes. Please review and sorry I didn't get the suggestions you mentioned for the description.

**Comment by JamesChenX** [comment]
@DevPJ9 I have submitted a commit based on your changes in https://github.com/turms-im/turms/tree/Issue_%231036, and you can pull it to your local and squash it with your changes into one commit, and push it to this PR forcibly so that we can merge it. Thanks.

**Comment by DevPJ9** [comment]
Hello @JamesChenX. I synced my personal git with your committed repo and all the changes have been synced here. Do I still need to do the rebasing?

**Comment by JamesChenX** [comment]
@DevPJ9 Thank you! Yes need to rebase, and you can do it in an easy way, just select multiple commits and squash them in IntelliJ IDEA.


Or you can use the commands mentioned in https://stackoverflow.com/questions/6934752/combining-multiple-commits-before-pushing-in-git

**Comment by DevPJ9** [comment]
Hello @JamesChenX I have done the changes please review.

**Comment by JamesChenX** [comment]
@DevPJ9 Merged. And thanks for your PR!

**Pull request merged**